test/test_puma_localhost_authority.rb

```diff
@@ -0,0 +1,117 @@
+# Nothing in this file runs if Puma isn't compiled with ssl support
+#
+# helper is required first since it loads Puma, which needs to be
+# loaded so HAS_SSL is defined
+require_relative "helper"
+require "localhost/authority"
+
+if ::Puma::HAS_SSL && !Puma::IS_JRUBY
+  require "puma/minissl"
+  require "puma/events"
+  require "net/http"
+
+  class SSLEventsHelper < ::Puma::Events
+    attr_accessor :addr, :cert, :error
+
+    def ssl_error(error, ssl_socket)
+      self.error = error
+      self.addr = ssl_socket.peeraddr.last rescue "<unknown>"
+      self.cert = ssl_socket.peercert
+    end
+  end
+
+
+  # net/http (loaded in helper) does not necessarily load OpenSSL
+  require "openssl" unless Object.const_defined? :OpenSSL
+
+
+  puts "", RUBY_DESCRIPTION, "RUBYOPT: #{ENV['RUBYOPT']}",
```

Comment:
This is not necessary because it's already logged in another SSL test file (can't remember which one)

See double output here: 
https://github.com/puma/puma/pull/2610/checks?check_run_id=2932850256#step:9:25